### ISSUE TITLE
[ISSUE] Issue with Issue with recognizing platform (32-bit, 64-bit) in terraform 

### ISSUE THREAD

**Issue by mohsas** [databricks/terraform-provider-databricks#4260]
Initializing provider plugins...
- Finding hashicorp/azurerm versions matching "4.10.0"...
- Finding databricks/databricks versions matching "1.58.0"...
- Installing hashicorp/azurerm v4.10.0...
- Installed hashicorp/azurerm v4.10.0 (signed by HashiCorp)
╷
│ Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/databricks/databricks v1.58.0 does not have
│ a package available for your current platform, windows_386.
│ 
│ Provider releases are separate from Terraform CLI releases, so not all
│ providers are available for all platforms. Other versions of this provider
│ may have different platforms supported.

The agent where pipeline runs is windows 64-bit.
It's like terraform doesn't recognize 64-bit windows platform.

Initializing provider plugins...
- Finding hashicorp/azurerm versions matching "4.10.0"...
- Finding databricks/databricks versions matching "1.58.0"...
- Installing hashicorp/azurerm v4.10.0...
- Installed hashicorp/azurerm v4.10.0 (signed by HashiCorp)
╷
│ Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/databricks/databricks v1.58.0 does not have
│ a package available for your current platform, windows_386.
│ 
│ Provider releases are separate from Terraform CLI releases, so not all
│ providers are available for all platforms. Other versions of this provider
│ may have different platforms supported.

**Comment by mohsas** [comment]
The configuration of providers as below:
required_providers {
    databricks = {
      source = "databricks/databricks"
      version = "1.58.0"
    }
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "4.10.0"
    }
  }

**Comment by alexott** [comment]
it's more a terraform problem - run `terraform -v` to check what architecture is supported by your TF

**Comment by mohsas** [comment]
Terraform v1.9.8
on windows_amd64 locally. Then sent to Agent in pipeline. The agent is 64-bit Windows Server

**Comment by mohsas** [comment]
It exactly mentions that system is 386 and it's not true:
Provider registry.terraform.io/databricks/databricks v1.58.0 does not have
│ a package available for your current platform, windows_386

**Comment by alexott** [comment]
Provider version is selected by Terraform, so it's not a provider issue. Check that your agent installs 64-bit TF version

**Comment by mohsas** [comment]
You are right the version of terraform in Agent is windows_386, changed it to amd64 and pushed pipeline and found that databricks tries to get checksum from github and it is not whitelisted in our organization.
│ Error: Failed to install provider
│ 
│ Error while installing databricks/databricks v1.58.0: could not query
│ provider registry for registry.terraform.io/databricks/databricks: failed
│ to retrieve authentication checksums for provider: the request failed after
│ 2 attempts, please try again later: Get
│ "[https://github.com/databricks/terraform-provider-databricks/releases/download/v1.58.0/terraform-provider-databricks_1.58.0_SHA256SUMS"](https://github.com/databricks/terraform-provider-databricks/releases/download/v1.58.0/terraform-provider-databricks_1.58.0_SHA256SUMS%22):
│ EOF

**Comment by alexott** [comment]
not a provider issue

**Comment by mohsas** [comment]
Thank you!.
